csharp/extractor/Semmle.Extraction.CSharp/SymbolExtensions.cs

```diff
@@ -193,8 +193,30 @@ public static void BuildArraySuffix(this IArrayTypeSymbol array, TextWriter trap
             trapFile.Write(']');
         }
 
+        private static void BuildAssembly(IAssemblySymbol asm, TextWriter trapFile, bool extraPrecise = false)
+        {
+            var assembly = asm.Identity;
+            trapFile.Write(assembly.Name);
+            trapFile.Write('_');
+            trapFile.Write(assembly.Version.Major);
+            trapFile.Write('.');
+            trapFile.Write(assembly.Version.Minor);
+            trapFile.Write('.');
+            trapFile.Write(assembly.Version.Build);
+            if (extraPrecise)
+            {
+                trapFile.Write('.');
+                trapFile.Write(assembly.Version.Revision);
+            }
+            trapFile.Write("::");
+        }
+
         static void BuildNamedTypeId(this INamedTypeSymbol named, Context cx, TextWriter trapFile, Action<Context, TextWriter, ITypeSymbol> subTermAction)
         {
+            bool prefixAssembly = true;
+            if (cx.Extractor.Standalone) prefixAssembly = false;
+            if (named.ContainingAssembly is null) prefixAssembly = false;
```

Comment:
I think it is OK to simplify to
```
var prefixAssembly = !cx.Extractor.Standalone && !(named.ContainingAssembly is null);
```

Reply:
Can we keep this as it is? The reason is because I know more cases will be added here in future pull requests.

Reply:
OK 👍